### PR TITLE
Persistent Project Id

### PR DESCRIPTION
**Pull request by hungnguyen94**
This feature adds the `currentProjectId` attribute to the User entity. That way, a user has the same projectId across all pages.   
A Liquibase changelog has been created to correctly add the column to the User table. An URI mapping has been created to fetch the currentProjectId(`api/account/currentproject`). To change the currentProjectId to `123`, call `api/account/update_currentproject?projectId=123`. 
Also added tests for the new resource class.   

I will add an AngularJS service to fetch this attribute in the frontend.

### CODE REVIEW

File: src/main/resources/config/liquibase/changelog/20160520223310_added_entity_column_User.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <changeSet id="20160520223310-1" author="Hung">
+        <addColumn tableName="jhi_user">
+            <column name="current_project_id" type="bigint">
+                <constraints nullable="true"/>
+            </column>
+        </addColumn>
+    </changeSet>
+</databaseChangeLog>
```
Comment (rubenwiersma) [inline src/main/resources/config/liquibase/changelog/20160520223310_added_entity_column_User.xml]: Shouldn't these files be ignored?

File: src/main/webapp/app/services/user/project-manager.service.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+(function() {
+    'use strict';
+    angular
+        .module('thefirstorderApp')
+        .factory('ProjectManager', ProjectManager);
+
+    ProjectManager.$inject = ['$http'];
+
+    function ProjectManager ($http) {
+        var resourceUrl =  'api/account/';
```
Comment (rubenwiersma) [inline src/main/webapp/app/services/user/project-manager.service.js]: This variable is not used

File: src/test/java/nl/tudelft/thefirstorder/web/rest/ProjectManagerResourceIntTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    private MockMvc restUserMockMvc;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        doNothing().when(mockMailService).sendActivationEmail((User) anyObject(), anyString());
+
+        ProjectManagerResource projectManagerResource = new ProjectManagerResource();
+//        ReflectionTestUtils.setField(projectManagerResource, "userRepository", userRepository);
```
Comment (rubenwiersma) [inline src/test/java/nl/tudelft/thefirstorder/web/rest/ProjectManagerResourceIntTest.java]: Commented code

File: src/test/java/nl/tudelft/thefirstorder/web/rest/ProjectManagerResourceIntTest.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+//            user.setCurrentProjectId(123L);
+//            return null;
+//        }).when(mockUserService).updateUserProjectId(anyLong());
+
+
+//        restUserMockMvc.perform(get("/api/account/currentproject")
+//                .accept(MediaType.APPLICATION_JSON))
+//                .andExpect(status().isNotFound());
+
+//        doCallRealMethod().when(mockUserService).updateUserProjectId(anyLong());
```
Comment (rubenwiersma) [inline src/test/java/nl/tudelft/thefirstorder/web/rest/ProjectManagerResourceIntTest.java]: Commented code could also be removed.

**Comment by rubenwiersma** [comment]
I've looked throught the code. Looks good, clean implementation of API structure.

File: src/main/resources/config/liquibase/changelog/20160520223310_added_entity_column_User.xml (hunk 1 above)
Comment (hungnguyen94) [inline src/main/resources/config/liquibase/changelog/20160520223310_added_entity_column_User.xml]: No, these are the changelogs for Liquibase. Liquibase refactors the database schema according to these changelogs files.

File: src/main/webapp/app/services/user/project-manager.service.js (hunk 2 above)
Comment (hungnguyen94) [inline src/main/webapp/app/services/user/project-manager.service.js]: Will remove!

File: src/test/java/nl/tudelft/thefirstorder/web/rest/ProjectManagerResourceIntTest.java (hunk 4 above)
Comment (hungnguyen94) [inline src/test/java/nl/tudelft/thefirstorder/web/rest/ProjectManagerResourceIntTest.java]: That's a test that might be improved, if anyone feels like doing that.

**Comment by hungnguyen94** [comment]
I removed the commented code.

**Pull request merged**